### PR TITLE
ci: upgrade notes

### PR DESCRIPTION
**Pull request by mattjoke**
* added check.yml workflow
* added .commitlintrc.json with specific rules
* changed README.md for consistency

### CODE REVIEW

File: .commitlintrc.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+			[
+				"core",
+				"openapi",
+				"cli",
+				"gui",
+				"engine",
+				"registrar",
+				"dispatcher",
+				"mvn",
+				"update"
```
Comment (mattjoke) [inline .commitlintrc.json:19]: This scope was added, so this PR wouldn't fail. It should be removed before merge.

**Comment by mattjoke** [comment]
Removed `--legacy-peer-deps` from workflows by updating `package.json` to the latest compatible version of the packages.

File: README.md [hunk 2]
```diff
@@ -1,39 +1,43 @@
-![](https://github.com/CESNET/perun/workflows/test%20build/badge.svg)
 
 <a href="https://perun-aai.org"><img style="float: left; position: relative;" src="https://raw.githubusercontent.com/CESNET/perun/master/perun-web-gui/src/main/webapp/img/logo.png"></a>
-## Identity & Access Management System ##
+## Identity & Access Management System
+
+[![conventional commits](https://img.shields.io/badge/semantic--release-conventional-e10079?logo=semantic-release)](https://www.conventionalcommits.org/)
+[![commitizen friendly](https://img.shields.io/badge/commitizen-friendly-brightgreen.svg)](https://commitizen.github.io/cz-cli/)
+[![CI status](https://github.com/cesnet/perun/actions/workflows/semantic-release.yml/badge.svg)](https://github.com/cesnet/perun/actions/workflows/semantic-release.yml)
```
Comment (melanger) [inline README.md:7]:
```suggestion
[![CI status](https://github.com/CESNET/perun/workflows/test%20build/badge.svg)](https://github.com/cesnet/perun/actions/workflows/maven.yml)
```

**Review by melanger — COMMENTED** [review]
Not a feature (`feat:`), it is ci/docs/chore, so use ci: or chore:

**Review by melanger — APPROVED** [review]

**Pull request merged**